server/cluster/coordinator.go

```diff
@@ -157,6 +162,33 @@ func (c *coordinator) patrolRegions() {
 	}
 }
 
+// checkSuspectKeyRanges would pop one suspect key range group
+// The regions of new version key range and old version key range would be placed into
+// the suspect regions map
+func (c *coordinator) checkSuspectKeyRanges() {
+	keyRangeGroup, success := c.cluster.PopOneSuspectKeyRange()
+	if !success {
+		return
+	}
+	regionMap := map[uint64]struct{}{}
+	f := func(regions []*core.RegionInfo) {
+		for _, region := range regions {
+			regionMap[region.GetID()] = struct{}{}
+		}
+	}
+	newKeyRegions := c.cluster.ScanRegions(keyRangeGroup[0], keyRangeGroup[1], patrolScanRegionLimit)
```

Comment:
The range to be very large. Use `patrolScanRegionLimit` may only scan a small part of the range.

Reply:
What about 1024 or 2048 for the limit?

Reply:
how about scan some regions and update the range so we can scan rest regions next scan?

Reply:
good idea, updated.